src/bokeh/models/ranges.py

```diff
@@ -407,7 +407,7 @@ class FactorRange(Range):
         available (e.g. server, notebook).
     """)
 
-    end = Readonly(Float, help="""
+    end = Readonly(Float, default=0, help="""
```

Comment:
Prior to this PR all readonly properties had an intrinsic default value, which usually doesn't make sense. So now they don't except for this, because `FactorRange` in bokehjs sets `start` and `end` during model initialization, which always results in 0, because that happens before sensible values can be computed. Due to how defaults' testing works, this requires setting the default value in bokeh. Ultimately this default will be removed, which will require changes to `FactorRange`'s initialization, which I don't want to get into here.

Reply:
Please open an issue for that and link back here.

Reply:
https://github.com/bokeh/bokeh/issues/12747